File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 1
        right = len(numbers)
        while left < right:
            currentSum = numbers[left-1] + numbers[right-1]
            if currentSum == target:
                return [left, right]
            elif currentSum < target:
                left += 1
            else: 
                right -= 1

File: Data Structures & Algorithms/spiral-matrix/submission-0.cs
public class Solution {
    public List<int> SpiralOrder(int[][] matrix) {
        int top = 0, bottom = matrix.Length - 1;
        int left = 0, right = matrix[0].Length - 1;
        List<int> res = new();
        while (top <= bottom && left <= right)
        {
            for (int i = left; i <= right; i++)
            {
                res.Add(matrix[top][i]);
            }
            top++;
            for (int i = top; i <= bottom; i++)
            {
                res.Add(matrix[i][right]);
            }
            right--;
            
            if (top <= bottom)
            {
                for (int i = right; i >= left; i--)
                {
                    res.Add(matrix[bottom][i]);
                }
                bottom--;
            }
            if (left <= right)
            {
                for (int i = bottom; i >= top; i--)
                {
                    res.Add(matrix[i][left]);
                }
                left++;
            }
        }
        return res;
    }
}
